### PR TITLE
fixes/ Semantic Html Issues

### LINKED ISSUE #2967: Semantic HTML issues in header section
### Increasing Access

Using proper semantic HTML allows screen readers to understand our content.

### Feature enhancement details

I found two issues relating to the header section while running an accessibility check:

1. We have a `<header>` element inside another `<header>` element, and multiple `<header>` elements on mobile.  We need to get this down to one.  We can change incorrect ones to either a React Fragment or a `<div>` or potentially remove them from the tree entirely.
2. We have a `<span>` as a direct child of a `<ul>` in the "login or signup" part of the nav bar.  Probably we change the `<span>or</span>` to `<li>or</li>`.  It would also be acceptable to have the entire "login or signup" as one list item, but making the "or" into its own list item is probably easier to implement.

We want to fix these HTML structures without changing the style.

### PR DESCRIPTION
**Pull request by Mubashirshariq**
Fixes #2967 

Changes:

I have verified that this pull request:

* [ ] has no linting errors (`npm run lint`)
* [ ] has no test errors (`npm run test`)
* [ ] is from a uniquely-named feature branch and is up to date with the  `develop` branch.
* [ ] is descriptively named and links to an issue number, i.e. `Fixes #123`

### CODE REVIEW

**Comment by Mubashirshariq** [comment]
@lindapaiste  plz review

File: client/modules/IDE/components/Header/index.jsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ const Header = (props) => {
   const project = useSelector((state) => state.project);
 
   return (
-    <header>
+    <>
```
Comment (lindapaiste) [inline client/modules/IDE/components/Header/index.jsx:12]: I wonder if it makes sense to keep the `<header>` here and remove the one in in the `NavBar` component instead?  It's subjective.  It depends on whether we consider the toolbar with the play button etc. to be part of the header or not.  I can see the arguments for both ways.

Comment (Mubashirshariq) [reply client/modules/IDE/components/Header/index.jsx:12]: In my perspective, centralizing the header in the NavBar component aligns with a cleaner and more cohesive design, emphasizing a unified user interface.

Comment (Mubashirshariq) [reply client/modules/IDE/components/Header/index.jsx:12]: But still if you want i will remove the one in the Navbar  one and keep this one.

**Comment by raclim** [comment]
Closing due to addressed by https://github.com/processing/p5.js-web-editor/pull/3130. I'm sorry that we couldn't get this in, but please feel free to check out our other issues!

**Pull request closed without merging**